controllers/gitrepository_controller_test.go

```diff
@@ -498,10 +498,12 @@ func TestGitRepositoryReconciler_reconcileSource_authStrategy(t *testing.T) {
 			}
 
 			r := &GitRepositoryReconciler{
-				Client:                      builder.Build(),
-				EventRecorder:               record.NewFakeRecorder(32),
-				Storage:                     testStorage,
-				features:                    features.FeatureGates(),
+				Client:        builder.Build(),
+				EventRecorder: record.NewFakeRecorder(32),
+				Storage:       testStorage,
+				features: map[string]bool{
+					features.ForceGoGitImplementation: false,
```

Comment:
By doing this, isn't it changing the previous configuration which had optimized clones true? I don't think it'll affect this test but maybe better to not alter the default behavior yet.
Same applies to other similar change below.